mlflow/store/dbfs_artifact_repo.py

```diff
@@ -114,10 +114,11 @@ def list_artifacts(self, path=None):
             return []
         dbfs_files = json_response.get('files', [])
         for dbfs_file in dbfs_files:
-            is_dir = dbfs_file['is_dir']
-            artifact_size = None if is_dir else dbfs_file['file_size']
             stripped_path = strip_prefix(dbfs_file['path'], artifact_prefix + '/')
-            infos.append(FileInfo(stripped_path, is_dir, artifact_size))
+            if stripped_path != path:
```

Comment:
Maybe add a comment on what this is doing -- namely that we're translating the underlying library's behavior of returning a single file to the API's expected behavior of returning an empty list. In fact, it may be clearer if you write it that way;
```
if stripped_path == path:
  return []
```